WeakAurasOptions/ConditionOptions.lua

```diff
@@ -2378,27 +2516,59 @@ local function addControlsForCondition(args, order, data, conditionVariable, tot
     end,
     func = function()
       if (data.controlledChildren) then
+        OptionsPrivate.Private.TimeMachine:StartTransaction()
         for id, reference in pairs(conditions[i].check.references) do
           local auraData = WeakAuras.GetData(id);
           local index = reference.conditionIndex;
           if (index > 1) then
-            local tmp = auraData[conditionVariable][reference.conditionIndex];
-            tremove(auraData[conditionVariable], reference.conditionIndex);
-            tinsert(auraData[conditionVariable], reference.conditionIndex - 1, tmp);
-            fixUpLinkedInFirstCondition(auraData[conditionVariable])
-            WeakAuras.Add(auraData);
+            local records = {}
+            tinsert(records, {
+              actionType = "swap",
+              uid = auraData.uid,
+              path = {"conditions"},
+              payload = {
+                index,
+                index - 1
+              }
+            })
+            if index == 2 and conditions[2].linked then
+              -- might be a bit confusing in this order?
+              -- but after the swap above, the second condition is now the first condition
+              tinsert(records, {
+                actionType = "set",
+                uid = auraData.uid,
+                path = {"conditions", 1, "linked"},
+                payload = false
+              })
+            end
+            OptionsPrivate.Private.TimeMachine:AppendMany(records)
```

Comment:
I think this is an example of two changes that depend on each other. And for this to work, the changes need to be reverse in the backwards apply.